### ISSUE TITLE
BUG regarding move.undo()

### ISSUE THREAD

**Issue by AlpacaMax** [brozeph/node-chess#71]
```js
> const chess = require('chess');
undefined
> const gameClient = chess.create({ PGN: true });
undefined
> let m = gameClient.move("e4");
undefined
> m = gameClient.move("c5");
undefined
> m.undo()
undefined
> gameClient.move("e5");
{
  move: {
    algebraic: 'e5',
    capturedPiece: null,
    castle: false,
    enPassant: false,
    postSquare: Square { file: 'e', piece: [Pawn], rank: 5 },
    prevSquare: Square { file: 'e', piece: null, rank: 4 } // Here's the bug. The prevSquare should be e7
  },
  undo: [Function (anonymous)]
}
```

**Comment by brozeph** [comment]
@AlpacaMax  - in this example, it looks like you move e4 then c5. Then c5 is undone, but you move the pawn from e4 to e5 in the next move after the undo... it feels like the bug here is that the board doesn't properly detect that it is the proper side's turn to move again. I'm building some unit tests to validate and begin fixing.

**Comment by brozeph** [comment]
@AlpacaMax - thank you for the great bug and repro steps... a patch is now available in `v0.6.0`!
